cpp/src/arrow/util/uri.h

```diff
@@ -68,6 +68,9 @@ class ARROW_EXPORT Uri {
   /// The URI path component.
   std::string path() const;
 
+  /// The Extension component for file
+  std::string extension() const;
```

Comment:
There is no concept of extension in URIs (the path may not represent a filename at all). Can't you just make it a private utility in substrait?